pom.xml

```diff
@@ -1490,5 +1490,28 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+    <!-- Workaround for https://bugs.openjdk.java.net/browse/JDK-8212233
+    TODO: likely just forbid 11.0.2 when 11.0.3 is out -->
+      <id>javadoc-crash-jdk-11.0.2</id>
+      <activation>
+        <property>
+          <name>java.version</name>
+          <value>11.0.2</value>
+        </property>
+      </activation>
+      <build>
+          <pluginManagement>
+              <plugins>
+                  <plugin>
+                      <artifactId>maven-javadoc-plugin</artifactId>
+                      <configuration>
+                          <source>${java.level}</source>
```

Comment:
I do not get it. Should this not just be the default?

```diff
diff --git a/pom.xml b/pom.xml
index 1fc3cdd..e44fcef 100644
--- a/pom.xml
+++ b/pom.xml
@@ -379,6 +379,7 @@
           <artifactId>maven-javadoc-plugin</artifactId>
           <version>3.0.1</version>
           <configuration>
+            <source>${java.level}</source>
             <quiet>true</quiet>
             <links>
               <link>http://javadoc.jenkins.io/</link>
```

Reply:
This is currently only needed and ever was, only on JDK 11.0.2. It started failing in CI recently once the JDK was bumped from 11.0.1  to 11.0.2, and this is deemed fixed in JDK 11.0.3, so I don't think it should be needed.

Reply:
The fact that builds _might_ have worked without this is interesting but my point is that this setting should have been configured on this mojo in all cases from the start, and that this was forgotten was only discovered thanks to this bug.

Reply:
As per the evaluation by Jonathan Gibbons:

> If this really is a JDK8-era program, use the javadoc option --source 8 or --release 8.